tests/compute/test_node_flow.py

```diff
@@ -0,0 +1,255 @@
+import backend as F
+import numpy as np
+import scipy as sp
+import dgl
+from dgl.node_flow import create_full_node_flow
+from dgl import utils
+import dgl.function as fn
+from functools import partial
+
+def generate_rand_graph(n, connect_more=False):
+    arr = (sp.sparse.random(n, n, density=0.1, format='coo') != 0).astype(np.int64)
+    # having one node to connect to all other nodes.
+    if connect_more:
+        arr[0] = 1
+        arr[:,0] = 1
+    g = dgl.DGLGraph(arr, readonly=True)
+    g.ndata['h1'] = F.randn((g.number_of_nodes(), 10))
+    g.edata['h2'] = F.randn((g.number_of_edges(), 3))
+    return g
+
+
+def create_mini_batch(g, num_hops):
+    seed_ids = np.array([0, 1, 2, 3])
+    seed_ids = utils.toindex(seed_ids)
+    sgi = g._graph.neighbor_sampling([seed_ids], g.number_of_nodes(), num_hops, "in", None)
+    assert len(sgi) == 1
+    return dgl.node_flow.NodeFlow(g, sgi[0])
+
+
+def check_basic(g, nf):
+    num_nodes = 0
+    for i in range(nf.num_layers):
+        num_nodes += nf.layer_size(i)
+    assert nf.number_of_nodes() == num_nodes
+    num_edges = 0
+    for i in range(nf.num_blocks):
+        num_edges += nf.block_size(i)
+    assert nf.number_of_edges() == num_edges
+
+    deg = nf.layer_in_degree(0)
+    assert F.array_equal(deg, F.zeros((nf.layer_size(0)), F.int64))
+    deg = nf.layer_out_degree(-1)
+    assert F.array_equal(deg, F.zeros((nf.layer_size(-1)), F.int64))
+    for i in range(1, nf.num_layers):
+        in_deg = nf.layer_in_degree(i)
+        out_deg = nf.layer_out_degree(i - 1)
+        assert F.asnumpy(F.sum(in_deg, 0) == F.sum(out_deg, 0))
+
```

Comment:
Move the section**s** below to `test_copy`?  I feel that some of the logics are duplicated.

Reply:
what logic is duplicated?

Reply:
L49-L66 duplicates L188-L198?

Reply:
you are right.